functions/Invoke-DbaLogShippingRecovery.ps1

```diff
@@ -0,0 +1,307 @@
+If this value is not supplied all databases will be recovered
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. To use:
+
+$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter. 
+
+Windows Authentication will be used if SqlCredential is not specified. SQL Server does not accept Windows credentials being passed as credentials. 	
+To connect as a different Windows user, run PowerShell as that user.
+
+.PARAMETER NoRecovery
+Allows you to choose to not restore the database to a functional state (Normal) in the final steps of the process.
+By default the database is restored to a functional state (Normal). 
+
+
+.NOTES 
+Author: Sander Stad (@sqlstad), sqlstad.com
+Requires: sysadmin access on SQL Servers
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Invoke-DbaLogShippingRecovery
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' 
+
+Recovers all the databases on the instance that are enabled for log shiping
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -SqlCredential $cred -Verbose
+
+Recovers all the databases on the instance that are enabled for log shiping using a credential
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -database 'db_logship' -Verbose
+
+Recovers the database "db_logship" to a normal status
+
+
+.EXAMPLE   
+db1, db2, db3, db4 | Invoke-DbaLogShippingRecovery -SqlServer 'server1' -Verbose
+
+Recovers the database db1, db2, db3, db4 to a normal status
+
+.EXAMPLE   
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -Verbose -WhatIf
+
+Shows what would happen if the command were executed.
+
+#>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+    param
+	(
+        [Parameter(Position=1, Mandatory=$true)]
+		[Alias("ServerInstance", "SqlInstance")]
+        [string]$SqlServer,
+        [Parameter(Position=2, Mandatory=$false, ValueFromPipeline=$true)]
+        [string[]]$Database = $null,
+        [Parameter(Position=3, Mandatory=$false)][object]$SqlCredential,
+        [Parameter(Position=4, Mandatory=$false)][switch]$NoRecovery
+	)
+
+    BEGIN
+    {
+        Write-Output "Attempting to connect to Sql Server.."
```

Comment:
Please replace all instances of Write-Output that you use to message the state/progress of your function with:
`Write-Message -Message "<Message Text here>" -Level 5 -Silent $Silent`
Choose an appropriate level:
2: Equivalent to Write-Host/Write-Information (By default, the users reads this)
5: Equivalent to Write-Verbose (By default, the user only reads this if specifying -Verbose as parameter)
8: Equivalent to Write-Debug (By default,, the user only sees this if specifying -Debug as parameter)
The smaller the value, the larger the probability it will appear on the screen of the user. It also ties in into the logging system (well ... as soon as I finish that later today).